openmdao/core/problem.py

```diff
@@ -1221,25 +1210,38 @@ def check_partials(self, out_stream=_DEFAULT_OUT_STREAM, includes=None, excludes
             The boolean 'rank_inconsistent' indicates if the derivative wrt a serial variable is
             inconsistent across MPI ranks.
         """
-        if self._metadata['setup_status'] < _SetupStatus.POST_FINAL_SETUP:
-            self.final_setup()
-
         model = self.model
 
+        # on master, we were doing this:
+        # if self._metadata['setup_status'] < _SetupStatus.POST_FINAL_SETUP:
+        #     self.final_setup()
+
+        # model.run_apply_nonlinear()
```

Comment:
It is interesting that we were only doing a run_apply, and I wonder if this caused some funny behavior. As a matter of habit, I always call run_model before check_partials, so I don't encounter problems, but that could be a learned behavior caused by this.